### ISSUE TITLE
ca65: add support for escapes in character constants

### ISSUE THREAD

**Issue by sidneycadot** [cc65/cc65#2571]
ca65 currently has (limited) support for C-like escape sequences in strings, by enabling the "string_escapes" feature.

It would be very helpful if there was a similar way to enable C-like escape characters in character constants.

This should probably be implemented in the assembler's scanner functions (src/ca65/scanner.c) in pretty much the same way that it's implemented in the compiler.

Not having easy access to the (target-specific) escape characters gives rise to bugs (eg #2562), and currently the solution for that kind of bug is hacky (#2565). Implementing this feature should allow us to clean up the solution implemented by #2565.

**Comment by mrdudz** [comment]
yes, and #2565 is a perfect example of a usecase :)

**Comment by sidneycadot** [comment]
See also issue #2581. This issue and #2581 should probably be fixed together.

**Comment by mrdudz** [comment]
Just to mention it here as well - the issue is really only "c-style escape sequences in character literals". Both strings and characters are already translated to the target character set, and escape sequences in strings do work.

**Comment by sidneycadot** [comment]
> [...] and escape sequences in strings do work.

They work partially. The escape sequence handling in the assembler's scanner handles a subset of the escape sequences handlerd by the compiler's scanner; those two should be equivalent I think.

**Comment by GorillaSapiens** [comment]
see also #2565 #2562.

**Comment by GorillaSapiens** [comment]
#2562 #2565 #2571 #2581 all related.

**Comment by GorillaSapiens** [comment]
see #2642

**Comment by mrdudz** [comment]
Yes, thanks again. My time budget is exhausted for this weekend, it would be great if you could have a look/check/ping me on issues/tickets that are related to this and need to be closed now (and finally this one too) :) @sidneycadot @GorillaSapiens

**Comment by mrdudz** [comment]
fine, so this can be closed as well ... #2581 needs some testing, but we can solve any problems (unlikely there are any) in a new ticket if needed
